check-ntpoffset/lib/check-ntpoffset.go

```diff
@@ -51,14 +51,20 @@ func run(args []string) *checkers.Checker {
 
 	var chkSt checkers.Status
 	var msg string
+	stratum, err := getNTPStratum()
+	if err != nil {
+		msg = fmt.Sprintf("ntp stratum not found offset is %f(actual), %f(warning threshold), %f(critical threshold)", math.Abs(offset), opts.Warn, opts.Crit)
+		return checkers.NewChecker(checkers.WARNING, msg)
```

Comment:
I think returning `UNKNOWN` result is better for this case, because the main purpose of the program (checking NTP offset) is not accomplished.

Reply:
I prefer to return `WARNING`. There are 3 cases that we should care.

1.  `offset` is able to retrieved, and synchronized to stratum
2. `offset` is able to retrieved, but failed to synchronized to stratum
3. failed to get `offset`

`WARNING` is suitable for case2, so that NTP stratum could be not synchronized but accessible state.

Reply:
But for simplify the problem, I agree you.

Reply:
I am concerning about that, if it returns WARNING for the case 2, interpretation of WARNING result would be ambiguous.
What do you think?

Reply:
For now, returns `UNKNOWN` in case failed to get NTP stratum.